File: server/order-service/src/main/java/com/sudo248/invoiceservice/controller/dto/CartDto.java
package com.sudo248.invoiceservice.controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CartDto {
    private String cartId;
    private Double totalPrice;
    private int totalAmount;
    private String status;
    private List<CartSupplierProductDto> cartSupplierProducts;
}

File: server/cart-service/src/main/kotlin/com/sudoo/cartservice/controller/dto/CartDto.kt
package com.sudoo.cartservice.controller.dto

import com.sudoo.cartservice.repository.entity.Cart

data class CartDto(
    val userId: String = "",
    val cartId: String = "",
    var totalPrice: Double = 0.0,
    var quantity: Int = 0,
    val status: String = "",
    var cartProducts: List<CartProductDto> = listOf()
)

fun CartDto.toCart(): Cart {
    return Cart(userId = this.userId, cartId = this.cartId, quantity = this.quantity, status = this.status)
}